README.md

```diff
@@ -1,7 +1,298 @@
+
+Maintaining such a collection of test project codebases can be a burden. For example, if you need to upgrade a third-party dependency, you will have to repeat the same steps in every test project codebase. There's also a risk of your projects getting out of sync.
+
+## How scenario-tester solves the problem
+
+With `scenario-tester`, you only need to maintain a small number of project templates. In the simplest case, it will be just one test app or package. For complex cases, several template projects may be used.
+
+`scenario-tester` lets you take a template project and customize it by creating *scenarios*: copies of the project containing unique changes. For example, each scenario could have a different version of an npm dependency added to the project. And then you can use your JS test suite of choice to run tests against each scenario.
+
+## Example usage
+
+Here's a very basic example:
+
+```js
+import { Scenarios } from 'scenario-tester';
+import { Project } from 'fixturify-project';
+import { module, test } from 'qunit';
+
+Scenarios
+
+  // Use a project from filesystem as a template
+  .fromProject(() => Project.fromDir(dirname(require.resolve('../test-app/package.json')), { linkDevDeps: true })))
+
+  // Define multiple scenarios, each scenario is derived from the template and has unique dependencies installed
+  .expand({
+    modern: async (project) => {
+      project.linkDevDependency('some-npm-package', { baseDir: __dirname, resolveName: 'some-npm-package' });
+      project.linkDevDependency('another-npm-package', { baseDir: __dirname, resolveName: 'another-npm-package' });
+    },
+
+    legacy: async function (project) => {
+      project.linkDevDependency('some-npm-package', { baseDir: __dirname, resolveName: 'some-npm-package--legacy' });
+      project.linkDevDependency('another-npm-package', { baseDir: __dirname, resolveName: 'another-npm-package--legacy' });
+    }
+  })
+
+  // Emit derivative projects to the filesystem
+  .forEachScenario(scenario => {
+
+    // Define your test suite
+    module(scenario.name, function (hooks) {
+      hooks.beforeEach(assert => {
+        // ...
+      });
+
+      test('output directories exist', async function () {
+        // ...
+      });
+
+      test('package.json is modified appropriately', async function () {
+        /// ...
+      });
+    });
+  });
+```
+
+Note: no new project codebases will be emitted to your filesystem until `forEachScenario` is executed. This lets you manipulate scenarios before emitting their codebases.
+
+## Creating and manipulating project codebases
+
+Instead of defining a template project from disk as shown above, you can do it programmatically.
+
+`scenario-tester` relies on [stefanpenner/node-fixturify-project](https://github.com/stefanpenner/node-fixturify-project) (npm package [fixturify-project](https://www.npmjs.com/package/fixturify-project)).
```

Comment:
I think we can link to one or the other github or npm? they link back to each other so it doesn't matter which one we link to here